### DISCUSSION TITLE
Package existence check

### DISCUSSION THREAD

**Discussion by clsty** [end-4/dots-hyprland#204] [General]
I've just noticed that yay will not exit non-zero when package not found, which paru will do.

Eg. hyprshade is an AUR package, while hyprshades is not.

`yay --noconfirm -S hyprshades; echo "Exit code=$?"`:

```plain
 -> No AUR package found for hyprshades
 there is nothing to do
Exit code=0
```

`paru --noconfirm -S hyprshades; echo "Exit code=$?"`:
```plain
:: Resolving dependencies...
error: could not find all required packages:
    hyprshades (target)
Exit code=1
```

Well, what does this mean?
It simply means that in installation script, if a package not exists, then that whole line in `dependencies.txt` will be skipped because the exit code is 0.

Considering that some packages may disappear in AUR, it's important to solve this problem by some ways, such as:
- Make yay exit 1 when package not found. (I still don't know how to achieve this.)
- Use paru instead (but I remembered that hyprland does not recommend it, right?)
- Check the existence of every AUR package before installing them (which will consume extra bandwidth for AUR server).
- Write a function to install AUR package using makepkg (maybe a bad idea idk)

<https://github.com/Jguer/yay/issues/430> said that was fixed(?), but according to my exp it wasn't...

EDIT: paru is unrecommended for installing `hyprland-git`, but not other packages.
From hyprland wiki:

> If you are using the AUR (hyprland-git) package, you will need to cleanbuild to update the package. Paru has been problematic with updating before, use Yay.

**Comment by clsty** [comment]
I've submitted a issue here: <https://github.com/Jguer/yay/issues/2346>

But better not count on it, because the author of yay seems not very active recently, and there're about one hundreds of issues opened.

**Comment by clsty** [comment]
I wrote a `checkdeps.sh` in <https://github.com/end-4/dots-hyprland/pull/205>, but this is only a workaround.

**Comment by clsty** [comment]
@end-4 Could you please help me test these commands and post the output? Thanks.

```bash
yay --noconfirm -S hyprshades; echo "yay: Exit code=$?"
paru --noconfirm -S hyprshades; echo "paru: Exit code=$?"
```

**Reply by end-4** [reply]
yay = 0
paru = 1
